docs/decisions/0002-lti-1p3-variables.rst

```diff
@@ -0,0 +1,123 @@
+variables required by LTI 1.3 launches.
+
+Decisions
+=========
+
+The following section go over required and optional variables in each supported claim in the LTI 1.3 implementation.
+This ADR should be updated when new claims are added.
+
+Deployment ID
+~~~~~~~~~~~~~
++---------------------------------------------------------+------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| LTI 1.3 variable name                                   | Open edX         | Reasoning                                                                                                                                                                                                                       |
++---------------------------------------------------------+------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| https://purl.imsglobal.org/spec/lti/claim/deployment_id | "1" (fixed value)| We're using single-tenant deployments, were each key is used for a single XBlock. Even when moving up to the course level, we'll can keep this setup since multi-tenancy will only be required for instance wide configuration. |
++---------------------------------------------------------+------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+
+See http://www.imsglobal.org/spec/lti/v1p3/#single-tenant-tool-registered-and-deployed-once for more details.
+
+Roles
+~~~~~
+The roles claim takes in a list of LTI 1.3 compliant roles from http://www.imsglobal.org/spec/lti/v1p3/#role-vocabularies.
+The claim name is: https://purl.imsglobal.org/spec/lti/claim/roles.
+
+The mappings from edX platform are shown in the table below:
+
++---------------+------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------+
+| Open edX role | LTI 1.3 Roles included                                                 | Reasoning                                                                                     |
++---------------+------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------+
+| guest         | 'http://purl.imsglobal.org/vocab/lis/v2/institution/person#Student'    | Guests and students only have permission to view and interact with the LTI content.           |
```

Comment:
I'm afraid I don't know what you mean by guest in the Open edX context? Is it an anonymous user?

Reply:
I think that when I did the implementation I wanted to cover all use cases (values that could come up), but I think it makes sense to not pass any roles to a `guest` user since it's anonymous.
We should probably return an empty array in that case, does that seems better?

Reply:
I think my confusion stemmed from "guest" being an Open edX role, since I haven't encountered that terminology before. 

I don't know how common this use case would be, but perhaps for public courses LTI can use a student role but otherwise it can be an empty array? Does that make sense? 

I'm just outlining the approach here not suggesting we develop this right now.

Reply:
I think we should go with the empty array if the user is a `guest` (or any other type of user not specified).